production/helm/loki/templates/_helpers.tpl

```diff
@@ -248,6 +253,11 @@ gcs:
   request_timeout: {{ .requestTimeout }}
   enable_http2: {{ .enableHttp2}}
 {{- end -}}
+{{- else if eq .Values.loki.storage.type "azure" -}}
+{{- with .Values.loki.storage.azure }}
+azure:
+{{- toYaml . }}
```

Comment:
What's the argument against following the same pattern as above, where we explicitly pull out the individual config fields that are valid?

What I like about this is it also encourages us to add a default section under `.Values.loki.storage.azure` so people know what options are available, since Loki only supports specific ones.

Would you be open to changing it to follow that pattern?